optaplanner-core/src/main/java/org/optaplanner/core/impl/localsearch/decider/multithreaded/MultiThreadedLocalSearchDecider.java

```diff
@@ -0,0 +1,407 @@
+                                logIndentation, moveThreadIndex, stepIndex, lastStepScore);
+                        try {
+                            // Don't consume an MoveEvaluationOperation until every moveThread took his ApplyStepOperation
+                            moveThreadBarrier.await();
+                        } catch (InterruptedException | BrokenBarrierException e) {
+                            Thread.currentThread().interrupt();
+                            break;
+                        }
+                    } else if (operation instanceof MoveEvaluationOperation) {
+                        MoveEvaluationOperation<Solution_> moveEvaluationOperation = (MoveEvaluationOperation<Solution_>) operation;
+                        int moveIndex = moveEvaluationOperation.getMoveIndex();
+                        if (stepIndex != moveEvaluationOperation.getStepIndex()) {
+                            throw new IllegalStateException("Impossible situation: the moveThread's stepIndex ("
+                                    + stepIndex + ") differs from the operation's stepIndex ("
+                                    + moveEvaluationOperation.getStepIndex() + ") with moveIndex ("
+                                    + moveIndex + ").");
+                        }
+                        Move<Solution_> move = moveEvaluationOperation.getMove().rebase(scoreDirector);
+                        if (!move.isMoveDoable(scoreDirector)) {
+                            logger.trace("{}            Move thread ({}) evaluation: step index ({}), move index ({}), not doable.",
+                                    logIndentation, moveThreadIndex, stepIndex, moveIndex);
+                            resultQueue.addUndoableMove(moveThreadIndex, stepIndex, moveIndex, move);
+                        } else {
+                            Score score = scoreDirector.doAndProcessMove(move, assertMoveScoreFromScratch);
+                            if (assertExpectedUndoMoveScore) {
+                                scoreDirector.assertExpectedUndoMoveScore(move, lastStepScore);
+                            }
+                            logger.trace("{}            Move thread ({}) evaluation: step index ({}), move index ({}), score ({}).",
+                                    logIndentation, moveThreadIndex, stepIndex, moveIndex, score);
+                            // Deliberately add to fail fast if there is not enough capacity (which is impossible)
+                            resultQueue.addMove(moveThreadIndex, stepIndex, moveIndex, move, score);
+                        }
+                    } else {
+                        throw new IllegalStateException("Unknown operation (" + operation + ").");
+                    }
+                    // TODO checkYielding();
+                }
+                logger.trace("{}            Move thread finished.");
```

Comment:
Is there a missing log parameter (logIndentation) here?

Reply:
Yes, thanks for reporting :)